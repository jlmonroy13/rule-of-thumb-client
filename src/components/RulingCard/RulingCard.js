import React from 'react';
import PropTypes from 'prop-types';

import Icon from 'components/Icon';

const RulingCard = ({
  name,
  imageUrl,
  startDate,
  category,
  description,
  upThumbs,
  downThumbs,
}) => (
  <div className="ruling-card" style={{ backgroundImage: `url(${imageUrl})` }}>
    <div className="ruling-card__mask" />
    <div className="ruling-card__body">
      <div className="position-relative">
        <span className="ruling-card__icon">
          <Icon name="thumb" className="ruling-card__thumb" />
        </span>
        <h2 className="ruling-card__title">{name}</h2>
      </div>
      <span className="text-small mb-3 d-block">
        <b>{startDate}</b> in {category}
      </span>
      <p className="ruling-card__description">{description}</p>
      <div className="d-flex align-items-center">
        <button className="ruling-card__thumb-btn mr-2">
          <Icon
            name="thumb"
            className="ruling-card__thumb ruling-card__thumb--up"
          />
        </button>
        <button className="ruling-card__thumb-btn mr-3">
          <Icon name="thumb" className="ruling-card__thumb" />
        </button>
        <button className="ruling-card__button">Vote now</button>
      </div>
    </div>
    <div className="ruling-card__footer">
      <div className="ruling-card__footer-percentage ruling-card__footer-percentage--up">
        <Icon name="thumb" className="ruling-card__footer-icon" />
        <span>{upThumbs}%</span>
      </div>
      <div
        className="ruling-card__footer-percentage"
        style={{ width: `${downThumbs}%` }}
      >
        <span className="mr-2">{downThumbs}%</span>
        <Icon name="thumb" />
      </div>
    </div>
  </div>
);

RulingCard.propTypes = {
  name: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  startDate: PropTypes.string.isRequired,
  category: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  upThumbs: PropTypes.number.isRequired,
  downThumbs: PropTypes.number.isRequired,
};

export default RulingCard;
