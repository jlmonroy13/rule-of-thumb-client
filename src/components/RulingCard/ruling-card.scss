
$class-name: '.ruling-card';

#{$class-name} {
  background-size: cover;
  background-repeat: no-repeat;
  height: 540px;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  position: relative;
  margin-bottom: 30px;

  &__mask {
    position: absolute;
    background-image:
      linear-gradient(
        to bottom,
        rgba(0, 0, 0, 0),
        rgba(0, 0, 0, 0.9)
      );
    top: 200px;
    bottom: 0;
    left: 0;
    width: 100%;
    z-index: 1;
  }

  &__body {
    z-index: 2;
    margin-bottom: 30px;
    padding: 0 35px;
    color: $white;
  }

  &__title {
    @include font-size(45);

    line-height: 45px;
    font-weight: 400;
    margin: 0;
  }

  &__description {
    line-height: 1rem;
  }

  &__icon {
    position: absolute;
    bottom: 5px;
    left: -35px;
  }

  &__thumb {
    @include font-size(32);

    padding: 5px;
    background-color: $secondary-color;

    &-btn {
      padding: 0;
      border: 2px solid $transparent;

      &:hover,
      &.is-active {
        border: 2px solid $white;
      }
    }

    &--up {
      background-color: $main-color;
      transform: rotate(180deg);
    }
  }

  &__button {
    color: $white;
    border: 1px solid $white;
    padding: 7px 15px;

    &:hover {
      transition: color 0.2s;
      background-color: $white;
      color: $body-color;
    }
  }

  &__footer {
    display: flex;
    color: $white;
    align-items: center;
    z-index: 2;

    &-percentage {
      @include font-size(29);

      display: flex;
      align-items: center;
      padding: 4px 0 4px 10px;
      background-color: rgba($secondary-color, 0.7);

      &--up {
        flex-grow: 1;
        background-color: rgba($main-color, 0.7);
      }
    }

    &-icon {
      transform: rotate(180deg);
      margin-right: 8px;
    }
  }
}
