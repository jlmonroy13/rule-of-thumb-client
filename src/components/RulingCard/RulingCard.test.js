/**
 *
 * Tests for RulingCard
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from '@testing-library/react';

import kanyePhoto from 'assets/images/west.jpg';

import RulingCard from './index';

const props = {
  name: 'Kanye West',
  imageUrl: kanyePhoto,
  startDate: '2020/02/25',
  category: 'Entertaiment',
  description:
    'Lorem ipsum dolor sit amet consectetur adipisicing elit. Consectetur, voluptates magnam excepturi libero ipsum ipsa eligendi voluptate praesentium.',
  upThumbs: 64,
  downThumbs: 36,
};

describe('<RulingCard />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<RulingCard {...props} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<RulingCard {...props} />);
    expect(firstChild).toMatchSnapshot();
  });
});
