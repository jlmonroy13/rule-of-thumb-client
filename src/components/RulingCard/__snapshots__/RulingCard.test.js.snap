// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RulingCard /> Should render and match the snapshot 1`] = `
<div
  class="ruling-card"
  style="background-image: url(west.jpg);"
>
  <div
    class="ruling-card__mask"
  />
  <div
    class="ruling-card__body"
  >
    <div
      class="position-relative"
    >
      <span
        class="ruling-card__icon"
      >
        <svg
          class="icon icon-thumb ruling-card__thumb"
          data-testid="icon-component"
        >
          <use
            xlink:href="#icon-thumb"
          />
        </svg>
      </span>
      <h2
        class="ruling-card__title"
      >
        Kanye West
      </h2>
    </div>
    <span
      class="text-small mb-3 d-block"
    >
      <b>
        2020/02/25
      </b>
       in 
      Entertaiment
    </span>
    <p
      class="ruling-card__description"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Consectetur, voluptates magnam excepturi libero ipsum ipsa eligendi voluptate praesentium.
    </p>
    <div
      class="d-flex align-items-center"
    >
      <button
        class="ruling-card__thumb-btn mr-2"
      >
        <svg
          class="icon icon-thumb ruling-card__thumb ruling-card__thumb--up"
          data-testid="icon-component"
        >
          <use
            xlink:href="#icon-thumb"
          />
        </svg>
      </button>
      <button
        class="ruling-card__thumb-btn mr-3"
      >
        <svg
          class="icon icon-thumb ruling-card__thumb"
          data-testid="icon-component"
        >
          <use
            xlink:href="#icon-thumb"
          />
        </svg>
      </button>
      <button
        class="ruling-card__button"
      >
        Vote now
      </button>
    </div>
  </div>
  <div
    class="ruling-card__footer"
  >
    <div
      class="ruling-card__footer-percentage ruling-card__footer-percentage--up"
    >
      <svg
        class="icon icon-thumb ruling-card__footer-icon"
        data-testid="icon-component"
      >
        <use
          xlink:href="#icon-thumb"
        />
      </svg>
      <span>
        64
        %
      </span>
    </div>
    <div
      class="ruling-card__footer-percentage"
      style="width: 36%;"
    >
      <span
        class="mr-2"
      >
        36
        %
      </span>
      <svg
        class="icon icon-thumb "
        data-testid="icon-component"
      >
        <use
          xlink:href="#icon-thumb"
        />
      </svg>
    </div>
  </div>
</div>
`;
